.ci/test.sh

```diff
@@ -75,6 +75,75 @@ if [[ $TASK == "if-else" ]]; then
     exit 0
 fi
 
+if [[ $TASK == "r-pkg" ]]; then
```

Comment:
I feel like it can be moved into a separate `.ci/test_r_package.sh` file, since it's self-contained, rather big and includes a lot of installation stuff.

Reply:
ok sure, I'd be happy to do that. I wasn't sure where we draw the line between keeping `test.sh` readable and minimizing the number of files to manage in `.ci/`

Reply:
Yeah, it's not so clear... I proposed that mostly because there are a lot of preparation stuff in the script which we used to keep in `setup.sh` file.

Reply:
moved everything into `.ci/test_r_package.sh` in https://github.com/microsoft/LightGBM/pull/2530/commits/dc95c332739a709ff926e1589b0ecceaefc8ebe6